snap/info.go

```diff
@@ -950,6 +951,109 @@ func (s *Info) DesktopPrefix() string {
 	return fmt.Sprintf("%s+%s", s.SnapName(), s.InstanceKey)
 }
 
+// DesktopFileIDs returns desktop-file-ids desktop plug attribute entries.
+// The desktop-file-ids attribute is optional so an empty list is returned if
+// the it is not found.
+//
+// Note: DesktopFileIDs doesn't check if the desktop plug is connected because
+// the desktop-file-ids attribute is controlled by an allow-installation rule.
+func (s *Info) DesktopFileIDs() ([]string, error) {
```

Comment:
maybe this should be named DesktopPlugFileIDs(), otherwise the difference between DesktopFilesFromMount() isn't very clear